ui/app/src/components/PagesWidget.tsx

```diff
@@ -0,0 +1,692 @@
+    }
+  ],
+  [
+    {
+      id: 'history',
+      label: 'History'
+    },
+    {
+      id: 'traslation',
+      label: 'Traslation'
+    }
+  ]
+];
+
+const itemsSelectedMenu = [
+  [
+    {
+      id: 'itemsSelected',
+      label: '',
+      type: 'text'
+    },
+    {
+      id: 'terminateSelection',
+      label: 'Terminate Selection'
+    }
+  ],
+  [
+    {
+      id: 'cut',
+      label: 'Cut'
+    },
+    {
+      id: 'copy',
+      label: 'Copy'
+    },
+    {
+      id: 'paste',
+      label: 'Paste'
+    },
+    {
+      id: 'duplicate',
+      label: 'Duplicate'
+    },
+    {
+      id: 'delete',
+      label: 'Delete'
+    }
+  ],
+  [
+    {
+      id: 'traslation',
+      label: 'Translation'
+    }
+  ]
+];
+
+interface PagesHeaderProps {
+  currentLocale: string;
+
+  setCurrentLocale(locale: string): void;
+}
+
+function PagesHeader(props: PagesHeaderProps) {
+  const classes = useStyles({});
+  const [menu, setMenu] = React.useState({
+    anchorEl: null,
+    type: null
+  });
+  const { currentLocale, setCurrentLocale } = props;
+
+  const localeList = [
+    {
+      id: 'en',
+      label: 'English, US (en)'
+    },
+    {
+      id: 'es',
+      label: 'Spanish, Spain (es)'
+    }
+  ];
+
+  const currentFlag = (locale: string) => {
+    switch (locale) {
+      case 'en': {
+        return <PhotoSizeSelectActualIcon className={classes.blackColor}/>
+      }
+      case 'es': {
+        return <PlaceRoundedIcon className={classes.blackColor}/>
+      }
+      default: {
+        return <PhotoSizeSelectActualIcon className={classes.blackColor}/>
+      }
+    }
+  };
+
+  const changeLocale = (locale: string) => {
+    setMenu({ ...menu, anchorEl: null });
+    setCurrentLocale(locale);
+  };
+
+  return (
+    <header className={classes.pagesHeader}>
+      <DescriptionOutlinedIcon/>
+      <Typography variant="subtitle1" className={classes.pagesHeaderTitle}>
+        Pages
+      </Typography>
+      <IconButton
+        aria-label="language select"
+        className={classes.icon}
+        onClick={(event) => setMenu({ ...menu, anchorEl: event.currentTarget, type: 'language' })}
+      >
+        {currentFlag(currentLocale)}
+      </IconButton>
+      <IconButton
+        aria-label="options"
+        className={classes.icon}
+        onClick={(event) => setMenu({ ...menu, anchorEl: event.currentTarget, type: 'options' })}
+      >
+        <MoreVertIcon className={classes.blackColor}/>
+      </IconButton>
+      <Menu
+        anchorEl={menu.anchorEl}
+        open={Boolean(menu.anchorEl)}
+        onClose={() => {
+          setMenu({ ...menu, anchorEl: null });
+        }}
+      >
+        {
+          menu.type === 'language' ? (
+            localeList.map(locale =>
+              <MenuItem key={locale.id} onClick={() => changeLocale(locale.id)}>{locale.label}</MenuItem>
+            )
+          ) : (
+            <MenuItem onClick={() => {
+            }}>
+              option1
```

Comment:
?

Reply:
this menu does have an options mockup on figma